### PR TITLE
Use yq to process YAML content

### PR DESCRIPTION
**Pull request by honnix**
## Description

`sed` has its limitation when processing structured data. Introducing `yq` to extract `tags` and `categories` is more robust.

A concrete case I had to deal with is YAML array defined in the format of:

```
tags:
- foo
- baar
- 42
```

The downside is pretty clear that users will need to install yet another tool. So please free feel to reshape this PR or reject it.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## How has this been tested

- [ ] I have run `bash ./tools/build.sh && bash ./tools/test.sh` (at the root of the project) locally and passed
- [x] I have tested this feature in the browser

### Test Configuration

- Browerser type & version: Chrome version 85.0.4183.102 (Official Build) (64-bit)
- Operating system: Ubuntu 20.04 kernel 5.4.0-47-generic
- Bundler version: Bundler version 2.1.4
- Ruby version: ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x86_64-linux]
- Jekyll version: jekyll 4.1.1

### Checklist

- [x] My code follows the [Google style guidelines](https://google.github.io/styleguide/)
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by cotes2020** [comment]
`yq` seems to be a better choice, but your PR mess up the Github Action CI workflow, and the delpoy hook, too.

**Comment by honnix** [comment]
Which part is "messing up"? The PR branch build is useful for some cases I think. I can remove that part to make this PR smaller.

File: .github/workflows/pages-deploy.yml.hook [hunk 1]
```diff
@@ -22,50 +29,9 @@ jobs:
...
-    steps:
-      - uses: actions/setup-ruby@v1
-        with:
-          ruby-version: '2.6.x'
-
-      - name: Checkout
-        uses: actions/checkout@v2
-        with:
-          fetch-depth: 0
+          brew install yq
```
Comment (cotes2020) [inline .github/workflows/pages-deploy.yml.hook:34]: Ubuntu did not have the `brew` tool

File: .github/workflows/pages-deploy.yml.hook [hunk 2]
```diff
@@ -22,50 +29,9 @@ jobs:
...
-
-    steps:
-      - uses: actions/setup-ruby@v1
-        with:
-          ruby-version: '2.6.x'
-
-      - name: Checkout
-        uses: actions/checkout@v2
-        with:
-          fetch-depth: 0
```
Comment (cotes2020) [inline .github/workflows/pages-deploy.yml.hook:68]: If `build-n-test` step is deleted, the test of the site with `baseurl` will fail (the htmlproofer's limitation)

**Review by cotes2020 — CHANGES_REQUESTED** [review]
In summary, the change of `_scripts/sh/create_pages.sh` is perfect, but `.github/workflows/ci.yml` lacks the installation of `yq`, and `.github/workflows/pages-deploy.yml.hook` has some shortcomings. If you want, please remove the modification of `pages-deploy.yml.hook` from this PR, I will submit a commit to improve it later, otherwise, I can modify your commit directly, but GitHub will not recognize that this PR is merged.

File: .github/workflows/pages-deploy.yml.hook (hunk 1 above)
Comment (honnix) [reply .github/workflows/pages-deploy.yml.hook:34]: It has. :)

File: .github/workflows/pages-deploy.yml.hook (hunk 2 above)
Comment (honnix) [reply .github/workflows/pages-deploy.yml.hook:68]: Interesting. I thought this was just a CI job name.

**Comment by honnix** [comment]
Yeah I can revert the change on CI part.

File: _scripts/sh/create_pages.sh [hunk 3]
```diff
@@ -25,19 +25,26 @@ _read_yaml() {
...
   local _yaml="$(_read_yaml "$1")"
-  echo "$_yaml" | grep "^tags *:" | sed "s/tags *: *//;s/\[//;s/\].*//;s/, */,/g;s/\"//g;s/'//g"
+  local _tags="$(echo "$_yaml" | yq r - "tags.*")"
+  local _tag="$(echo "$_yaml" | yq r - "tag")"
+
+  if [[ -n $_tags ]]; then
+    echo "$_tags"
+  elif [[ -n $_tag ]]; then
+    echo "$_tag"
+  fi
```
Comment (cotes2020) [inline _scripts/sh/create_pages.sh:47]: Sorry, there is one more thing, Jekyll only supports the `tags` variable, so this part can be reduced to:

```diff
-local _tag="$(echo "$_yaml" | yq r - "tag")"
-
-if [[ -n $_tags ]]; then
-  echo "$_tags"
-elif [[ -n $_tag ]]; then
-  echo "$_tag"
-fi
+echo "$_tags"
```

Comment (honnix) [reply _scripts/sh/create_pages.sh:47]: Hmm, this [doc](https://jekyllrb.com/docs/posts/#tags) says both though.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -29,6 +29,10 @@ jobs:
         with:
           ruby-version: '2.6.x'
 
+      - name: install tooling
```
Comment (honnix) [inline .github/workflows/ci.yml:32]: Not sure why this step is not picked up by CI. :thinking:

Comment (honnix) [reply .github/workflows/ci.yml:32]: nvm I commented too early..

File: .github/workflows/pages-deploy.yml.hook (hunk 1 above)
Comment (cotes2020) [reply .github/workflows/pages-deploy.yml.hook:71]: Oh, I used to think the `brew` was only available on macOS 😂

File: _scripts/sh/create_pages.sh (hunk 3 above)
Comment (cotes2020) [reply _scripts/sh/create_pages.sh:47]: OK, my bad, I'm out of date :)

File: .github/workflows/pages-deploy.yml.hook (hunk 1 above)
Comment (honnix) [reply .github/workflows/pages-deploy.yml.hook:71]: Yeah me too, until I got a PC recently and installed Linux. :)

**Pull request merged**